src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -46,9 +54,16 @@ public TElement[] ToArray()
 
             TElement[] array = new TElement[count];
             int[] map = SortedMap(buffer);
```

Comment:
Okay, I just checked and `Return` throws if you give it null, so that check would be needed.